File: platform/polyglot/saucelabs_api/py/saucelabs_api/modules/jobs.py
"""
Jobs Module — Sauce Labs API Client

Manage test execution history (VDC & RDC).

Endpoints:
    GET /rest/v1/{username}/jobs          - List jobs
    GET /rest/v1.1/{username}/jobs/{id}   - Get job details
"""

from __future__ import annotations

from typing import Any

from ..errors import SaucelabsValidationError
from ..logger import create_logger

log = create_logger("saucelabs_api", "jobs")


class JobsModule:
    """Manage Sauce Labs test jobs."""

    def __init__(self, client: Any) -> None:
        self._client = client
        self._logger = log

    async def list(self, params: dict[str, Any] | None = None) -> list[Any]:
        """List test jobs for the configured user.

        Parameters
        ----------
        params:
            Optional dict with: limit (default 25), skip, from, to.
        """
        params = params or {}
        username = self._client.username
        if not username:
            raise SaucelabsValidationError("username is required to list jobs")

        query: dict[str, Any] = {"limit": params.get("limit", 25), "format": "json"}
        if "skip" in params:
            query["skip"] = params["skip"]
        if "from" in params:
            val = params["from"]
            if not isinstance(val, int) or val < 0:
                raise SaucelabsValidationError('"from" must be a positive integer (Unix timestamp)')
            query["from"] = val
        if "to" in params:
            val = params["to"]
            if not isinstance(val, int) or val < 0:
                raise SaucelabsValidationError('"to" must be a positive integer (Unix timestamp)')
            query["to"] = val

        self._logger.debug("listing jobs", {"username": username, "params": query})
        return await self._client.get(f"/rest/v1/{username}/jobs", params=query)

    async def get(self, job_id: str) -> dict[str, Any]:
        """Get details for a specific job."""
        username = self._client.username
        if not username:
            raise SaucelabsValidationError("username is required to get job")
        if not job_id:
            raise SaucelabsValidationError("job_id is required")

        self._logger.debug("getting job", {"username": username, "job_id": job_id})
        return await self._client.get(f"/rest/v1.1/{username}/jobs/{job_id}")

File: platform/polyglot/saucelabs_api/mjs/src/modules/jobs.mjs
/**
 * Jobs Module — Sauce Labs API Client
 *
 * Manage test execution history (VDC & RDC).
 *
 * Endpoints:
 *   GET /rest/v1/{username}/jobs          - List jobs
 *   GET /rest/v1.1/{username}/jobs/{id}   - Get job details
 */

import { create } from '../logger.mjs';
import { SaucelabsValidationError } from '../errors.mjs';

const log = create('saucelabs-api', import.meta.url);

export class JobsModule {
  /**
   * @param {import('../client.mjs').SaucelabsClient} client
   */
  constructor(client) {
    this._client = client;
    this._logger = log;
  }

  /**
   * List test jobs for the configured user.
   *
   * @param {object} [params={}]
   * @param {number} [params.limit=25] - Number of jobs to return
   * @param {number} [params.skip] - Number of jobs to skip
   * @param {number} [params.from] - Unix timestamp start filter
   * @param {number} [params.to] - Unix timestamp end filter
   * @returns {Promise<Array>} List of job objects
   */
  async list(params = {}) {
    const username = this._client.username;
    if (!username) {
      throw new SaucelabsValidationError('username is required to list jobs');
    }

    const queryParams = { limit: params.limit ?? 25, format: 'json' };
    if (params.skip !== undefined) queryParams.skip = params.skip;
    if (params.from !== undefined) {
      if (!Number.isInteger(params.from) || params.from < 0) {
        throw new SaucelabsValidationError('"from" must be a positive integer (Unix timestamp)');
      }
      queryParams.from = params.from;
    }
    if (params.to !== undefined) {
      if (!Number.isInteger(params.to) || params.to < 0) {
        throw new SaucelabsValidationError('"to" must be a positive integer (Unix timestamp)');
      }
      queryParams.to = params.to;
    }

    this._logger.debug('listing jobs', { username, params: queryParams });
    return this._client.get(`/rest/v1/${username}/jobs`, { params: queryParams });
  }

  /**
   * Get details for a specific job.
   *
   * @param {string} jobId - The job identifier
   * @returns {Promise<object>} Job details
   */
  async get(jobId) {
    const username = this._client.username;
    if (!username) {
      throw new SaucelabsValidationError('username is required to get job');
    }
    if (!jobId) {
      throw new SaucelabsValidationError('jobId is required');
    }

    this._logger.debug('getting job', { username, jobId });
    return this._client.get(`/rest/v1.1/${username}/jobs/${jobId}`);
  }
}
